plugins/netctl/cmdimpl/cmd_utils.go

```diff
@@ -31,55 +31,82 @@ import (
 	"time"
 )
 
-// FindIPForNodeName will find an ip address that corresponds to the passed
-// in nodeName
-func FindIPForNodeName(nodeName string) string {
-	cfg := &etcd.ClientConfig{
-		Config: &clientv3.Config{
-			Endpoints: []string{"127.0.0.1:32379"},
-		},
-		OpTimeout: 1 * time.Second,
-	}
-	// Create connection to etcd.
-	logger := logrus.DefaultLogger()
-	logger.SetLevel(logging.FatalLevel)
-	db, err := etcd.NewEtcdConnectionWithBytes(*cfg, logger)
-	if err != nil {
-		fmt.Printf("Could not connect to Etcd to get IP address for node '%s', error '%s'\n", nodeName, err)
-		os.Exit(1)
-	}
-	itr, err := db.ListValues("/vnf-agent/contiv-ksr/allocatedIDs/")
-	if err != nil {
-		fmt.Printf("Could not get node info from Etcd for node '%s'\n", nodeName)
-		return ""
+type clusterNodeInfo map[string]*node.NodeInfo
+
+var (
+	nodeInfo    clusterNodeInfo
+	bytesBroker *etcd.BytesConnectionEtcd
+)
+
+func getEtcdBroker() *etcd.BytesConnectionEtcd {
+	if bytesBroker == nil {
+		etcdCfg := etcd.ClientConfig{
+			Config: &clientv3.Config{
+				Endpoints: []string{etcdLocation},
+			},
+			OpTimeout: 1 * time.Second,
+		}
+
+		logger := logrus.DefaultLogger()
+		logger.SetLevel(logging.ErrorLevel)
+
+		var err error
+		var db *etcd.BytesConnectionEtcd
+		if db, err = etcd.NewEtcdConnectionWithBytes(etcdCfg, logger); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		bytesBroker = db
 	}
-	for {
-		kv, stop := itr.GetNext()
-		if stop {
-			break
+
+	return bytesBroker
+}
+
+func getClusterNodeInfo() clusterNodeInfo {
+	if (nodeInfo == nil) || (len(nodeInfo) == 0) {
```

Comment:
No need to make extra check for nil, `len(nodeInfo)` will return 0 even when it's nil.